src/Linux/Core/Impl/OS/UnixProcess.cs

```diff
@@ -36,9 +35,17 @@ public class UnixProcess : IProcess {
             }
         }
 
-
         public void Kill() {
-            _process.Kill();
+            // This is needed because broker user cannot kill process running as another user.
+            KillProcess(_ps, _process.Id);
+        }
+
+        private static void KillProcess(IProcessServices ps, int pid) {
+            ProcessStartInfo psi = new ProcessStartInfo();
+            psi.FileName = UnixProcessServices.RunAsUserBinPath;
+            psi.Arguments = "-k {pid}";
```

Comment:
What I meant is that then we won't need the `-k` switch at all, so RAU code remains single-purpose (launching elevated processes).